Kernel/Prekernel/init.cpp

```diff
@@ -105,6 +121,9 @@ extern "C" [[noreturn]] void init()
         auto& kernel_program_header = kernel_program_headers[i];
         if (kernel_program_header.p_type != PT_LOAD)
             continue;
+        // we require that the ELF kernel is a DROW
+        if (!is_drow_kernel(kernel_program_header))
+            halt();
```

Comment:
(same as with the `halt()` above.)

Reply:
In the Prekernel environment there's no PANIC facility or debug console currently.